canvas/gradient.go

```diff
@@ -3,49 +3,101 @@ package canvas
 import (
 	"image"
 	"image/color"
-	"image/draw"
 	"math"
-)
 
-const (
-	// GradientDirectionVertical defines a vertically traveling gradient
-	GradientDirectionVertical GradientDirection = iota
-	// GradientDirectionHorizontal defines a horizontally traveling gradient
-	GradientDirectionHorizontal
-	// GradientDirectionCircular defines a gradient traveling radially from a center point outward
-	GradientDirectionCircular
+	"fyne.io/fyne"
 )
 
-// GradientDirection defines a starting or stopping location
-type GradientDirection int
+// Gradient describes a gradient fade between two colors.
+type Gradient interface {
```

Comment:
I'm not sure about the interface here - CanvasObjects are structs due to their concrete nature.
I see that users can create &LinearGradient{} or &RadialGradient{} so maybe it's not an issue - did you consider this in the design?